sklearn/model_selection/tests/test_search.py

```diff
@@ -1433,7 +1431,12 @@ def test_grid_search_failing_classifier():
     # error in this test.
     gs = GridSearchCV(clf, [{'parameter': [0, 1, 2]}], scoring='accuracy',
                       refit=False, error_score=0.0)
-    assert_warns(FitFailedWarning, gs.fit, X, y)
+    warning_message = (
+        "Estimator fit failed. The score on this train-test partition "
+        "for these parameters will be set to 0.000000."
```

Comment:
I wonder if the number of decimals can change depending on the platform, version of Python and so on. To avoid platform specific failures, I would rather use a regexp (I haven't tested it):

```suggestion
        "for these parameters will be set to 0.0.*."
```

Reply:
I checked the suggestion code in my environment and it works correctly.